src/main/java/com/dreamhorizon/enchantments/listener/EntityListener.java

```diff
@@ -37,11 +35,10 @@ public void onPlayerAttackPlayer(EntityDamageByEntityEvent event) {
         if (event.getDamager() instanceof LivingEntity && event.getEntity() instanceof LivingEntity) {
             LivingEntity damager = (LivingEntity) event.getDamager();
             LivingEntity damaged = (LivingEntity) event.getEntity();
-            EntityEquipment equipmentSlot = damager.getEquipment();
-            if (equipmentSlot == null) {
+            ItemStack attackItem = getAttackItem(damager);
+            if (attackItem == null) {
                 return;
             }
-            ItemStack attackItem = equipmentSlot.getItemInMainHand();
             if (attackItem.getType() == Material.AIR
```

Comment:
Why didn't you add the AIr Check to GetAttackItem, make it return null if it is air.